packages/material-ui/src/Modal/TrapFocus.js

```diff
@@ -31,15 +31,21 @@ function TrapFocus(props) {
   }, []);
   const handleRef = useForkRef(children.ref, handleOwnRef);
 
-  // ⚠️ You may rely on React.useMemo as a performance optimization, not as a semantic guarantee.
-  // https://reactjs.org/docs/hooks-reference.html#usememo
-  React.useMemo(() => {
-    if (!open || typeof window === 'undefined') {
-      return;
-    }
-
+  const prevOpenRef = React.useRef();
+  React.useEffect(() => {
+    prevOpenRef.current = true;
```

Comment:
```suggestion
    prevOpenRef.current = open;
```
Was this supposed to be `open`?